app/components/page.tsx

```diff
@@ -1,14 +1,38 @@
+import { ChevronLeft, X } from 'lucide-react';
+import React, { useEffect } from 'react';
+import {
+  LinkWithViewTransition,
+  type ViewTransitionLinkProps,
+} from '~/lib/transitions';
 import { cn } from '~/lib/utils';
 
 interface PageProps extends React.HTMLAttributes<HTMLDivElement> {
   children: React.ReactNode;
 }
 
 export function Page({ children, className, ...props }: PageProps) {
+  // vh-100 includes the space behind the search bar on mobile browsers.
```

Comment:
what does it mean `space behind the search bar`? also is this happening on all browsers?

I would also extract this to some custom hook

Reply:
The problem is described [here](https://css-tricks.com/the-trick-to-viewport-units-on-mobile/). TLDR: viewport height is handled inconsistently across browsers, but seems like it only happens on safari.

It seems like using `h-dvh` as pointed out in [this stack exchange](https://stackoverflow.com/questions/61308575/tailwind-h-screen-doesn-t-work-properly-on-mobile-devices) works

See comparison below, note that without those styles the form is centered relative to the entire screen including the search/toolbar at the bottom

With `h-dvh`:
![image|5%](https://github.com/user-attachments/assets/a3f21254-c33f-4fc1-91eb-cd936a15de54)

Without:
![image|50%](https://github.com/user-attachments/assets/97415b52-28ba-479d-b55b-fb0e174d269e)

Reply:
great. I love when we can get rid of useEffect :)